src/exe/model.cc

```diff
@@ -124,22 +126,53 @@ void ReadFileCameraLocations(const std::string& ref_images_path,
     ref_image_names.push_back(image_name);
     ref_locations.push_back(camera_position);
   }
+
+  if (ref_is_gps) {
+    GPSTransform gps_transform(GPSTransform::WGS84);
+    if (alignment_type != "enu") {
+      std::cout << "\n Converting Alignment Coordinates from GPS (lat/lon/alt) "
+                   "to ECEF. \n";
+      ref_locations = gps_transform.EllToXYZ(ref_locations);
+    } else {
+      std::cout << "\n Converting Alignment Coordinates from GPS (lat/lon/alt) "
+                   "to ENU. \n";
+      ref_locations = gps_transform.EllToENU(ref_locations);
+    }
+  } else {
+    std::cout << "\n Cartesian Alignment Coordinates extracted (MUST NOT BE "
+                 "GPS coords!). \n";
+  }
 }
 
 void ReadDatabaseCameraLocations(const std::string& database_path,
+                                 const bool ref_is_gps,
+                                 const std::string& alignment_type,
                                  std::vector<std::string>& ref_image_names,
                                  std::vector<Eigen::Vector3d>& ref_locations) {
   Database database(database_path);
   auto images = database.ReadAllImages();
-  std::vector<Eigen::Vector3d> gps_locations;
-  GPSTransform gps_transform(GPSTransform::WGS84);
   for (const auto image : images) {
     if (image.HasTvecPrior()) {
       ref_image_names.push_back(image.Name());
-      gps_locations.push_back(image.TvecPrior());
+      ref_locations.push_back(image.TvecPrior());
     }
   }
-  ref_locations = gps_transform.EllToXYZ(gps_locations);
+
+  if (ref_is_gps) {
+    GPSTransform gps_transform(GPSTransform::WGS84);
+    if (alignment_type != "enu") {
+      std::cout << "\n Converting Alignment Coordinates from GPS (lat/lon/alt) "
```

Comment:
nit: Remove the beginning/trailing spaces in the logged text here and elsewhere. Thanks.

Reply:
Thanks, it's done